### PR TITLE
Hamilton cov file

### PR DESCRIPTION
**Pull request by mayabrandi**
### This PR adds | fixes:
- adds a new epp for generation of hamilton file for covid preps
- adds a new group command to the cli: hamilton, that groups the three epps: barcode_file make_kapa_csv and sars_cov2_prep_file 
- moved get_index_well from epp make_kapa_csv to get/fields.py
- fixes a bug in the sort well function in files/manage_csv_files.py

### Before deploy:
- make sure all usage of  EPPs: barcode_file and make_kapa_csv are being updated with the new cli command.

### Review:
- [x] Code approved by @Mropat 
- [x] Tests executed by @cesve @mayabrandi 
- [x] "Merge and deploy" approved by @mayabrandi 

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg_lims/files/manage_csv_files.py [hunk 1]
```diff
@@ -48,16 +48,25 @@ def sort_csv(file: Path, columns: List[str], well_columns: List[str] = []):
...
     columns: list of columns to sort on.
 
-    well_columns: Optional. Sub set of columns. Will be sorted by second field:
+    well_columns: Optional. Sub set of columns! Will be sorted by second field:
                 (A1, B1, C1, A2, B2, C2), not (A1, A2, B1, B2, C1, C2).
     """
 
-    df = pd.read_csv(file.absolute(), delimiter=",")
+    data = pd.read_csv(file.absolute(), delimiter=",")
+    print(data)
```
Comment (Mropat) [inline cg_lims/files/manage_csv_files.py:56]: :stuck_out_tongue:

File: cg_lims/EPPs/files/hamilton/base.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# commands
+from cg_lims.EPPs.files.hamilton.normalization_file import barcode_file
+from cg_lims.EPPs.files.hamilton.make_kapa_csv import make_kapa_csv
+from cg_lims.EPPs.files.hamilton.sars_cov2_prep_file import sars_cov2_prep_file
+
+
+@click.group(invoke_without_command=True)
+@click.pass_context
+def hamilton(ctx):
+    """Main entry point of file commands"""
```
Comment (Mropat) [inline cg_lims/EPPs/files/hamilton/base.py:12]: doncstring doesnt match

File: cg_lims/files/manage_csv_files.py [hunk 3]
```diff
@@ -48,16 +48,25 @@ def sort_csv(file: Path, columns: List[str], well_columns: List[str] = []):
     """Sort csv.
     columns: list of columns to sort on.
 
-    well_columns: Optional. Sub set of columns. Will be sorted by second field:
+    well_columns: Optional. Sub set of columns! Will be sorted by second field:
                 (A1, B1, C1, A2, B2, C2), not (A1, A2, B1, B2, C1, C2).
     """
 
-    df = pd.read_csv(file.absolute(), delimiter=",")
+    data = pd.read_csv(file.absolute(), delimiter=",")
```
Comment (Mropat) [inline cg_lims/files/manage_csv_files.py:55]: more specific variable names are generally better!

**Comment by cesve** [comment]
# Verification Test

The new script for creating the Covid file has been added to the following steps:

Work Flow: SARS-CoV-2 v1
Protocol: Library Prep (Cov) v1
Step: Library Preparation (Cov) and Pooling and Clean-up (Cov)

## Step Library Preparation (Cov)

The file was created successfully and contained the correct information. The samples are not pooled in this step but they still have a destination well in the file. That information is not used but as we wanted to use the same script for both LIMS steps we decided to include that information. The file works in the protocol on Hamilton instrument.

Screenshot 2021-06-15 at 21 24 56

[92-1454623-sars-cov2-hamilton-prep.txt](https://github.com/Clinical-Genomics/cg_lims/files/6662774/92-1454623-sars-cov2-hamilton-prep.txt)

## Pooling and Clean-up (Cov)

The file was created successfully and contained the correct information. The indecies are not used in this step of the protocol or in this LIMS step. Therefor a "-" is written instead of an index well position. The file works in the protocol on Hamilton instrument.

Screenshot 2021-06-16 at 14 01 06

[92-1454644-sars-cov2-hamilton-prep.txt](https://github.com/Clinical-Genomics/cg_lims/files/6662821/92-1454644-sars-cov2-hamilton-prep.txt)

**Comment by mayabrandi** [comment]
Checking witch configurations need to be updated after merge

```
(cg_lims) mayabrmbp:clinical_EPPs maya.brandi$ python clinical_EPPs/check_configuration.py -s barcode-file
INFO:AUTOMATION: Searching for 'barcode-file'.
INFO:Button: Hamilton Normalization File
INFO:Bash string: bash -c -l "conda activate cg_lims &&  lims -c /home/glsai/.genologics.yaml epps -l {compoundOutputFileLuid0} -p {processLuid}  files make-hamilton-barcode-file -f {compoundOutputFileLuid4}  -b 'Volume H2O (ul)' -v 'Sample Volume (ul)' "
INFO:Processes: ['1357'] 

INFO:AUTOMATION: Searching for 'barcode-file'.
INFO:Button: Hamilton Normalization File
INFO:Bash string: bash -c -l "conda activate cg_lims &&  lims -c /home/glsai/.genologics.yaml epps -l {compoundOutputFileLuid0} -p {processLuid}  files make-hamilton-barcode-file -f {compoundOutputFileLuid3} -b 'Volume H2O (ul)' -v 'Volume of sample (ul)' -p"
INFO:Processes: ['1365'] 

(cg_lims) mayabrmbp:clinical_EPPs maya.brandi$ python clinical_EPPs/check_configuration.py -s make-kapa-csv
INFO:AUTOMATION: Searching for 'make-kapa-csv'.
INFO:Button: Sorted File for Hamilton
INFO:Bash string: bash -c -l "conda activate cg_lims && lims -c /home/glsai/.genologics.yaml epps -l {compoundOutputFileLuid0} -p {processLuid}  files  make-kapa-csv -f {compoundOutputFileLuid1} -n 'Aliquot samples for enzymatic fragmentation TWIST v2' --extension '.txt'"
INFO:Processes: ['1362', '1364'] 

```

**Pull request merged**